ForumProject/notifications/signals.py

```diff
@@ -44,42 +45,68 @@ def create_notifications(instances, follow_status_change=True):
         return Notification.objects.bulk_create(notifications)
     return []
 
-# Signal to create a notification when an investor starts following a project or changes a share
+def record_and_email_notifications(instance, follow_status_change=True):
+    """
+    Create Notification records and send email notifications for a given InvestorProject instance.
+
+    This function creates Notification records in the database and sends email notifications
+    to CustomUsers associated with the startup linked to the InvestorProject instance.
+
+    Parameters:
+    - instance (InvestorProject): The InvestorProject instance to process.
+    - follow_status_change (bool): Whether the trigger is a change in the following status
+      (default: True). If False, it is considered a change in the subscription share.
+    """
+    #trigger recording of notifications to the database
+    notifications = create_notifications(instance, follow_status_change=follow_status_change)
+    
+    for notification in notifications:
+        # Get all CustomUsers' emails associated with this startup
+        project = instance.project
+        startup = project.startup
+        user_startups = UserStartup.objects.filter(startup=startup)
```

Comment:
This line is run inside a loop for each notification. This could lead to performance issues due to repeated database queries. If possible, this query should be moved outside the loop or optimized to fetch all necessary data in fewer queries

Reply:
Done